File: RedSheet.DbEntities/Models/vNanoProjectNegotiableRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vNanoProjectNegotiableRecords", Schema = "dbo")]
	[Serializable]
	public partial class vNanoProjectNegotiableRecord  
	{
 
	[Required] 
	[MaxLength(1000)]
	public string ldo { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string mdo { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int NanoProjectNegotiableId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectModuleId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Requirement { get; set; } 

	public vNanoProjectNegotiableRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-nano-project-negotiable-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vNanoProjectNegotiableRecord {
    constructor(vNanoProjectNegotiableRecord?: vNanoProjectNegotiableRecord )  {
        let properties = [ "ldo", "mdo", "nanoProjectNegotiableId", "projectModuleId", "requirement",];
        for (let property of properties)
            if (vNanoProjectNegotiableRecord && vNanoProjectNegotiableRecord[property])
                this[property] = vNanoProjectNegotiableRecord[property];
    }
 
    @required()
    @maxLength(1000)
	ldo : string =   undefined;
 
    @required()
    @maxLength(1000)
	mdo : string =   undefined;
 
	nanoProjectNegotiableId : number =   0 ;
 
    @range(1,2147483647)
	projectModuleId : number =   undefined;
 
    @required()
    @maxLength(1000)
	requirement : string =   undefined;


}
